articles/cognitive-services/Bing-Web-Search/quickstarts/nodejs.md

```diff
@@ -22,102 +22,145 @@ Use this quickstart to make your first call to the Bing Web Search API and recei
+// HOWTO: set an environment variable https://is.gd/zu55jA
+const SUBSCRIPTION_KEY = process.env['AZURE_SUBSCRIPTION_KEY']
+if (!SUBSCRIPTION_KEY) {
+  throw new Error('Missing the AZURE_SUBSCRIPTION_KEY environment varable')
+}
 ```
 
-## Define variables
 
-A few variables must be set before we can continue. Confirm that the `host` and `path` are valid and replace the `subscriptionKey` value with a valid subscription key from your Azure account. Feel free to customize the search query by replacing the value for `term`.
+
+## Create a function to make the request
+
+This function will make a secure GET request, saving the search query as a query parameter in the path.
+We use `encodeURIComponent` to escape any invalid characters, and we pass our subscription key in a header.
+The callback receives a
+[response](https://nodejs.org/dist/latest-v10.x/docs/api/http.html#http_class_http_serverresponse),
+subscribing to the `data` event in order to aggregate the JSON body,
+the `error` event to log any issues,
+and the `end` event so it knows when it has the full body.
+At that time, it will print the interesting headers and the body.
+You can play with the colors and set the depth to suit your preference,
+a depth of `1` gives a nice summary of the response.
+
 
 ```javascript
-// Replace with a valid subscription key.
-let subscriptionKey = 'enter key here';
-
-/*
- * Verify the endpoint URI. If you
- * encounter unexpected authorization errors, double-check this host against
- * the endpoint for your Bing Web search instance in your Azure dashboard.  
- */
-let host = 'api.cognitive.microsoft.com';
-let path = '/bing/v7.0/search';
-let term = 'Microsoft Cognitive Services';
-
-// Validate the subscription key.
-if (subscriptionKey.length === 32) {
-    bing_web_search(term);
-} else {
-    console.log('Invalid Bing Search API subscription key!');
-    console.log('Please paste yours into the source code.');
+function bingWebSearch(query) {
+  https.get({
+    hostname: 'api.cognitive.microsoft.com',
+    path:     '/bing/v7.0/search?q=' + encodeURIComponent(query),
+    headers:  { 'Ocp-Apim-Subscription-Key': SUBSCRIPTION_KEY },
+  }, res => {
+    let body = ''
+    res.on('data', part => body += part)
+    res.on('end', () => {
+      for (var header in res.headers) {
+        if (header.startsWith("bingapis-") || header.startsWith("x-msedge-")) {
+          console.log(header + ": " + res.headers[header])
+        }
+      }
+      console.log('\nJSON Response:\n')
+      console.dir(JSON.parse(body), { colors: false, depth: null })
+    })
+    res.on('error', e => {
+      console.log('Error: ' + e.message)
+      throw e
+    })
+  })
 }
 ```
 
-## Create a response handler
+## Get the query
 
-Create a handler to stringify and parse the response. The `response_handler` is called each time a request is made to the Bing Web Search API, as you'll see in the next section.
+Here, we'll look at the program's arguments to find the query.
```

Comment:
Minor recommendation for the next few lines: 

Let's look at the program's arguments to find the query. The first argument is the path to node, the second is our filename, and the third is your query. If the query is absent, a default query of "Microsoft Cognitive Services" is used.